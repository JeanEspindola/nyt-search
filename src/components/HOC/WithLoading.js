import React from 'react';
import './WithLoading.css';

const WithLoading = loading => Component => props => (
  props[loading]
    ? <div className="loader" />
    : <Component {...props} />
);

export default WithLoading;
