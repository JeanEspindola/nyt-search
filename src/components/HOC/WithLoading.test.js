import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import WithLoading from './WithLoading';

describe(' WithLoading', () => {
  let Component;
  let WrapperComponent;

  it('renders WithLoading when isLoading true - without crashing', () => {
    WrapperComponent = WithLoading('loading')(Component);
    const wrapper = shallow(<WrapperComponent loading />);
    expect(wrapper.length).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders WithLoading when isLoading false - without crashing', () => {
    Component = () => (
      <div>Component</div>
    );
    WrapperComponent = WithLoading('loading')(Component);

    const wrapper = shallow(<WrapperComponent />);
    expect(wrapper.length).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
