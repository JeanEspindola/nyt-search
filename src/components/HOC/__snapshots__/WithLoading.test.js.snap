// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` WithLoading renders WithLoading when isLoading false - without crashing 1`] = `<Component />`;

exports[` WithLoading renders WithLoading when isLoading true - without crashing 1`] = `
<div
  className="loader"
/>
`;
